### PR TITLE
Add CompanyExportCountry model

### PR DESCRIPTION
**Pull request by benpbenp**
* Ignore these objects during company merge (replicates current
behaviour)
* Add set_user_edited_export_countries method to perform logic when user
sets these countries
* Add get_active_future_export_countries method to query the new model
and return a list of active countries of all sources.

### Description of change

Add a new model which can record future interest countries from a variety of sources.
This model is populated from the old future_interest_country m2m using a INSERT INTO... SELECT.

The two main sources of information are 'user' and 'external'. User represents those countries manually entered by the user on the front-end. External represents those countries that only came in from an import from the data science pipeline.

The key difference is that we want the user to be able to delete countries that came in from data science, and have those deletions stick when the import is refreshed. On the other hand, data science imports should only be able to add new countries, not delete user-entered countries or override any that users have deleted.

### Checklist

* [x] Has a new newsfragment been created? Check [changelog/README.md](https://github.com/uktrade/data-hub-leeloo/blob/master/changelog/README.md) for instructions
* [ ] Do any added or updated endpoints appear in the API documentation? See [docs/Maintaining the API documentation.md](https://github.com/uktrade/data-hub-leeloo/blob/develop/docs/Maintaining&#32;the&#32;API&#32;documentation.md) for more details
* [ ] Have any relevant search models been updated?
* [ ] Have any relevant fixtures (`fixtures/test_data.yaml`) been updated?
* [ ] Have any relevant select-/prefetch-related field lists in the views and search apps been updated?
* [ ] Has the admin site been updated (for new models, fields etc.)?
* [ ] Has the README been updated (if needed)?

### CODE REVIEW

File: changelog/company/company-export-country.db.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+A new table company_companyexportcountry was created.
```
Comment (elcct) [inline changelog/company/company-export-country.db.md]: The company_companyexportcountry could be wrapped with `, so it looks like this `company_companyexportcountry`.

File: datahub/company/models/company.py [hunk 2]
```diff
@@ -284,6 +285,78 @@ def companies_house_data(self):
             except CompaniesHouseCompany.DoesNotExist:
                 return None
 
+    def get_active_future_export_countries(self):
+        """
+        Get a list of unique countries for which there are active (non-deleted)
+        CompanyExportCountr
+        y Objects
```
Comment (elcct) [inline datahub/company/models/company.py]: Unnecessary line break.

File: datahub/company/models/company.py [hunk 3]
```diff
@@ -284,6 +285,78 @@ def companies_house_data(self):
...
+        update and delete CompanyExportCountry objects, as necessary.
+
+        Any existing CompanyExportCountries not in countries will be marked as deleted,
+        and have the 'user' source removed. If 'user' is the only source,
+        then the object will be completely removed from DB.
+        """
+        # Set to keep track of which input countries have been dealt with
+        countries = set(countries)
+
+        for cec in self.unfiltered_export_countries.select_related('country'):
```
Comment (elcct) [inline datahub/company/models/company.py]: I am not sure if `cec` is a good name, maybe we could have something more meaningful?

File: datahub/company/test/factories.py [hunk 4]
```diff
@@ -190,3 +200,28 @@ class ArchivedContactFactory(ContactFactory):
...
+
+    @factory.lazy_attribute
+    def disabled_on(self):
+        """
+        If you pass disabled=True to the factory, then a date in the past
+        will be generated and saved to the disabled_on field.
+        """
+        if self.disabled:
+            return factory.Faker('past_datetime', tzinfo=utc).generate()
+        else:
```
Comment (elcct) [inline datahub/company/test/factories.py]: Is this `else` needed here?

File: datahub/company/models/company.py [hunk 5]
```diff
@@ -284,6 +285,78 @@ def companies_house_data(self):
...
+            changed = False
+            if cec.country in countries:
```
Comment (elcct) [inline datahub/company/models/company.py]: The body of the `for` seems to have quite complicated logic. Is there a way to simplify this?

File: datahub/company/test/factories.py (hunk 4 above)
Comment (benpbenp) [reply datahub/company/test/factories.py]: no, no it's not...

File: datahub/company/models/company.py (hunk 3 above)
Comment (benpbenp) [reply datahub/company/models/company.py]: changed in other PR

File: datahub/company/models/company.py (hunk 2 above)
Comment (benpbenp) [reply datahub/company/models/company.py]: fixed in other PR

File: changelog/company/company-export-country.db.md (hunk 1 above)
Comment (benpbenp) [reply changelog/company/company-export-country.db.md]: ok fixed

**Comment by benpbenp** [comment]
Replaced by
#2181
#2182

**Pull request closed without merging**